pkg/broker/ingress/multi_topic_decouple_sink_test.go

```diff
@@ -220,6 +228,157 @@ func TestMultiTopicDecoupleSink(t *testing.T) {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "test-source",
+					},
+				},
+			},
+			hasTrigger: true,
+		},
+		{
+			name: "broker with target with non-matching filter",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "some-random-source",
+					},
+				},
+			},
+			hasTrigger: false,
+		},
+		{
+			name: "broker with multiple targets with one matching target filter",
+			brokerTargets: map[string]*config.Target{
+				"non_matching_target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "some-random-source",
+					},
+				},
+				"non_matching_target_2": {
+					FilterAttributes: map[string]string{
+						"source": "some-random-other-source",
+					},
+				},
+				"matching_target_3": {
+					FilterAttributes: map[string]string{
+						"type": eventType,
+					},
+				},
+			},
+			hasTrigger: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := logtest.TestContextWithLogger(t)
+			psSrv := pstest.NewServer()
+			defer psSrv.Close()
+			psClient := createPubsubClient(ctx, t, psSrv)
+
+			testBrokerConfig := &config.TargetsConfig{
+				Brokers: map[string]*config.Broker{
+					"test_ns_1/test_broker_1": {
+						DecoupleQueue: DecoupleQueue,
+						Targets:       test.brokerTargets,
+					},
+				},
+			}
+
+			brokerConfig := memory.NewTargets(testBrokerConfig)
+			sink := NewMultiTopicDecoupleSink(ctx, brokerConfig, psClient, pubsub.DefaultPublishSettings)
+
+			event := createTestEvent(uuid.New().String())
+			fmt.Printf("Event type: %s\n", event.Type())
+
+			hasTrigger := sink.hasTrigger(context.Background(), event)
+			if hasTrigger != test.hasTrigger {
+				t.Errorf("Sink says event has trigger %t which should be %t", hasTrigger, test.hasTrigger)
+			}
+		})
+	}
+}
+
+func TestMultiTopicDecoupleSinkSendChecksFilter(t *testing.T) {
+	filterCalled := false
+	orig := EventFilterFunc
+	defer func() { EventFilterFunc = orig }()
+	EventFilterFunc = func(ctx context.Context, attrs map[string]string, event *event.Event) bool {
+		filterCalled = true
+		return true
+	}
+
+	ctx := logtest.TestContextWithLogger(t)
+	psSrv := pstest.NewServer()
+	defer psSrv.Close()
+	psClient := createPubsubClient(ctx, t, psSrv)
+
+	testBrokerConfig := &config.TargetsConfig{
+		Brokers: map[string]*config.Broker{
+			"test_ns_1/test_broker_1": {
+				DecoupleQueue: &config.Queue{Topic: "test_topic_1", State: config.State_READY},
+				Targets:       map[string]*config.Target{"target_1": {}},
+			},
+		},
+	}
+
+	brokerConfig := memory.NewTargets(testBrokerConfig)
+
+	testTopic := "test_topic_1"
+	topic := psClient.Topic(testTopic)
+	if exists, err := topic.Exists(ctx); err != nil {
+		t.Fatal(err)
+	} else if !exists {
+		if topic, err = psClient.CreateTopic(ctx, testTopic); err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
Not sure if this complexity is needed here, since you always create a new server, is there any chance that there will be topic already in the test server? If so you should handle `Exists` properly. Otherwise you can safely call CreateTopic.

Reply:
I took this block of code from 'TestMultiTopicDecoupleSink'. Seems like the correct way to ensure the test is setup properly.